### PR TITLE
copr: Add vectorize FieldInt/FieldReal/FieldString

### PR DESCRIPTION
**Pull request by ty666**
PCP #5751

###  What have you changed?

Add vectorize FieldInt/FieldReal/FieldString

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

### CODE REVIEW

**Comment by ty666** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/93

File: components/tidb_query/src/rpn_expr/impl_string.rs [hunk 1]
```diff
@@ -239,6 +239,38 @@ pub fn locate_binary_3_args(
     }
 }
 
+#[rpn_fn(varg, min_args = 1)]
+#[inline]
+pub fn field_int(args: &[&Option<Int>]) -> Result<Option<Int>> {
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_string.rs]: I think you can write

```rust
#[rpn_fn(varg, min_args = 1)]
#[inline]
pub fn field<T: Evaluable + PartialEq>(args: &[&Option<T>]) -> Result<Option<Int>> { .. }
```

directly. You can assign it in the mapping using `ScalarFuncSig::FieldInt => field_fn_meta::<Int>()`

Comment (ty666) [reply components/tidb_query/src/rpn_expr/impl_string.rs]: ok

**Comment by ty666** [comment]
@sre-bot /run-integration-copr-test copr-test=pr/93

**Comment by breeswish** [comment]
Thanks a lot! Looks like field is not implemented for the **non-vectorized** framework. Could you implement that first?

**Comment by ty666** [comment]
> Thanks a lot! Looks like field is not implemented for the **non-vectorized** framework. Could you implement that first?

The `field` **non-vectorized** function has been implemented.

https://github.com/tikv/tikv/blob/06099157f916124377c046f122dec9dafa6096a1/components/tidb_query/src/expr/builtin_string.rs#L397-L442

**Comment by breeswish** [comment]
@ty666 Okay :)

**Pull request merged**